### PR TITLE
Type safe navigation

### PR DESCRIPTION
**Pull request by ComicSAS**
### Description
- Migrating from string routes to type safe navigation

### Checklist
- [x] I have ensured this PR does not contain any breaking changes for existing functionality;
- [x] I have added automated tests that cover the new behavior and removed obsolete tests and code;

### Tested On
- [ ] Emulator
- [x] Real device

### CODE REVIEW

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt [hunk 1]
```diff
@@ -85,7 +85,9 @@ class AiStableDiffusionActivity : AppCompatActivity() {
 
             LaunchedEffect(backStackEntry) {
                 if (!viewModel.state.value.isShowSplash) return@LaunchedEffect
-                if (backStackEntry?.destination?.route != Constants.ROUTE_SPLASH) {
+                backStackEntry?.let { entry ->
+                    val route = entry.destination.route ?: ""
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt:89]: What is the value of `route` in this case?

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/navigation/router/main/MainRouterImpl.kt [hunk 2]
```diff
@@ -19,9 +20,12 @@ internal class MainRouterImpl : MainRouter {
     }
 
     override fun navigateToOnBoarding(source: LaunchSource) {
-        effectSubject.onNext(NavigationEffect.Navigate.RouteBuilder("${Constants.ROUTE_ONBOARDING}/${source.ordinal}") {
+        effectSubject.onNext(NavigationEffect.Navigate.RouteBuilder(
+            "",
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/navigation/router/main/MainRouterImpl.kt:24]: Why there is empty string here ?

Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/navigation/router/main/MainRouterImpl.kt:24]: Will be changed in the future.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt (hunk 1 above)
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/activity/AiStableDiffusionActivity.kt:89]: route=com.shifthackz.aisdv1.presentation.navigation.NavigationRoute.Splash
route=com.shifthackz.aisdv1.presentation.navigation.NavigationRoute.Onboarding/{source}
route=com.shifthackz.aisdv1.presentation.navigation.NavigationRoute.GalleryDetail/{itemId}

**Review by ShiftHackZ — APPROVED** [review]

**Pull request merged**